### PR TITLE
feat(x/ecocredit): backport basket codegen

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

+ copied basets/v1 proto from `master`
+ added generated spec - now in `x/basket/spec` (it didn't work in otherway - need to fix it in some configuration later on)
+ updated proto/regen/ecocredit/v1alpha1/types.proto  as in #731 
+ change go package path -> don't use `v1` as package name.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/basket/spec/protobuf.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
+ <!-- This file is auto-generated. Please do not modify it yourself. -->
```
Comment (aaronc) [inline x/basket/spec/protobuf.md:1]: We're not creating an `x/basket` module

**Review by aaronc — COMMENTED** [review]
Where's the generated code?

**Review by aaronc — APPROVED** [review]
as long as @technicallyty and @ryanchristo are okay with the `go_package` change let's merge this.

would be good to understand the x/basket/spec thing

File: x/basket/spec/protobuf.md (hunk 1 above)
Comment (robert-zaremba) [reply x/basket/spec/protobuf.md:1]: bug in protocgen.sh -> https://github.com/regen-network/regen-ledger/blob/release/v2.2.x/scripts/protocgen.sh#L43
Will crerate a task for that

Comment (aaronc) [reply x/basket/spec/protobuf.md:1]: okay, it's probably not something we really need to worry about

**Review by ryanchristo — APPROVED** [review]
Looks good to me.

**Pull request merged**